### PR TITLE
WIP: add groups controller

### PR DESCRIPTION
**Pull request by malaffoon**
Another small change whilst i avoid the big refactor ... this adds the groups controller, and a method on the import service to deal with file uploads without loading the whole thing into memory.

### CODE REVIEW

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/auth/SbacUser.java [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
     private final List<Permission> permissions;
 
-    SbacUser(final String username,
-             final String password,
-             final Collection<? extends GrantedAuthority> authorities,
-             final String uid,
-             final String sbacUuid,
-             final TenancyChain tenancyChain,
-             final Collection<Permission> permissions) {
+    private SbacUser(final String username,
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/auth/SbacUser.java]: fix warning and cmd-alt-L side effects

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/auth/SecurityConfigurer.java [hunk 2]
```diff
@@ -88,6 +88,7 @@ public void configure(final HttpSecurity http) throws Exception {
                         .antMatchers(contentRoot + "/**").hasAuthority(DataLoadAuthority)
                         .antMatchers("/imports/**").hasAuthority(DataLoadAuthority)
                         // student groups end-points are treated differently; require GROUP_WRITE
+                        .antMatchers("/groups/imports/resubmit").authenticated()
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/auth/SecurityConfigurer.java]: tests actually found a problem

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java [hunk 3]
```diff
@@ -87,6 +92,56 @@ public RdwImport importContent(final SbacUserDetails user,
         return rdwImport;
     }
 
+    @Override
+    public RdwImport importContent(final SbacUserDetails user, final MultipartFile file, final ImportContent content, final String contentType, final Map<String, String> metadata) {
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: Duplicates the existing importContent method but deals with the file without pulling it all into memory.

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java [hunk 4]
```diff
@@ -87,6 +92,56 @@ public RdwImport importContent(final SbacUserDetails user,
...
+        rdwImport = repository.create(RdwImport.builder()
+                .content(content)
+                .contentType(contentType)
+                .digest(digest)
+                .status(ImportStatus.ACCEPTED)
+                .creator(user.getUsername())
+                .batch(metadata == null ? null : metadata.get("batch"))
+                .build());
+
+        source.submitContent(digest.getBytes(), content, contentType, rdwImport.getId());
```
Comment (malaffoon) [inline import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: for now, put the digest in as the body of the rabbit message; i'm still fiddling with the "best" way to build this message

File: import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java (hunk 3 above)
Comment (agorina) [reply import-service/src/main/java/org/opentestsystem/rdw/ingest/service/DefaultImportService.java]: There seems to be some overlap between both `importContent` methods. Maybe a helper for the shared functionality?

**Pull request merged**